### PR TITLE
Wikileaks Search Module and Util Class

### PR DESCRIPTION
**Pull request by prince-7**
Adding Wikileaks module and Util class https://github.com/saeeddhqan/Maryam/issues/163

### CODE REVIEW

File: core/util/wikileaks.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		""" wikileaks.org search engine
+
+			q		: query for search
+			limit	: Number of Pages
+
+		"""
+
+		self.framework = main.framework
+		self.q = q
+		self.agent = 'Mozilla/5.0 (X11; Linux x86_64; rv:86.0) Gecko/20100101 Firefox/86.0'
```
Comment (saeeddhqan) [inline core/util/wikileaks.py:33]: Do we need to use a specific agent?

File: modules/search/wikileaks.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+def module_api(self):
+	query = self.options['query']
+	limit = self.options['limit']
+	run = self.wikileaks(query, limit)
+	run.run_crawl()
+
+	output = {'results': []}
+	links = run.links_with_data
+
+	for item in links:
```
Comment (saeeddhqan) [inline modules/search/wikileaks.py:40]: output['results'] = links isn't better?

**Review by saeeddhqan — COMMENTED** [review]
everything looks good. test the code with max_attempts and let me know

File: core/util/wikileaks.py (hunk 1 above)
Comment (prince-7) [reply core/util/wikileaks.py:33]: No, not really

File: modules/search/wikileaks.py (hunk 2 above)
Comment (prince-7) [reply modules/search/wikileaks.py:40]: the output contains both links and titles of results 
```
self._links_with_data.append({
				'title': result.find_all('a')[0].text,
				'link': result.find_all('a')[0]['href']
				})
```

File: core/util/wikileaks.py [hunk 3]
```diff
@@ -38,26 +38,27 @@ def __init__(self, q, limit=1):
 		self._links_with_data = []
 
 	def run_crawl(self):
-		page = 1
```
Comment (prince-7) [inline core/util/wikileaks.py]: Should I replace page with max_attempts everywhere? also the max_attempts need to start with 1

Comment (prince-7) [reply core/util/wikileaks.py]: actually, the page was the user-supplied parameter, which signifies the number of pages the user wants to search, I am adding it again.

Comment (saeeddhqan) [reply core/util/wikileaks.py]: the user-supplied parameter: self.limit?

Comment (prince-7) [reply core/util/wikileaks.py]: yes, but I have handled it https://github.com/saeeddhqan/Maryam/pull/166/commits/fb9a2dd47edc503292221297f75c34bc08c78391

File: modules/search/wikileaks.py (hunk 2 above)
Comment (saeeddhqan) [reply modules/search/wikileaks.py:40]: the output format off links_with_data is not good. it should be something like this: [{'title': '....', 'link': '...}, ...] or [(title, link), (title, link), ...]
Also for the summer, we should organize/integrate the sources to have a consistent and similar result format.

Comment (prince-7) [reply modules/search/wikileaks.py:40]: well the format is like this  ```[{'title': '....', 'link': '...}, ...]``` only, links_with_data is a list.

Comment (prince-7) [reply modules/search/wikileaks.py:40]: The result format can be something like ```[{'link': '...', 'key': 'value', ....}, {'link': '...', 'key': 'value', ....}, ......]```

**Pull request merged**